utils.lua

```diff
@@ -104,9 +104,19 @@ end
 -- process the scores and obtain the ranks
 -- input: scores for all options, ground truth positions
 function utils.computeRanks(scores, gtPos)
-    local gtScore = scores:gather(2, gtPos);
-    local ranks = scores:gt(gtScore:expandAs(scores));
-    ranks = ranks:sum(2) + 1;
+    -- simply sort according to scores if ground truth not available
+    local ranks;
+    if gtPos then
+        gtPos = gtPos:view(-1, 1);
+        local gtScore = scores:gather(2, gtPos);
+        ranks = scores:gt(gtScore:expandAs(scores));
+        ranks = ranks:sum(2) + 1;
+    else
+        sorted, ranks = scores:sort(2)
+        for i = 1, scores:size(1) do
```

Comment:
I am not sure if this is doing the right thing. A unit test below with the output.

``` 
  1 require 'torch'
  2
  3 scores = torch.FloatTensor(10, 3):uniform();
  4
  5 sorted, ranks = scores:sort(2)
  6     for i = 1, scores:size(1) do
  7         ranks[i] = ranks[i]:index(1, ranks[i]);
  8     end
  9 print(scores)
 10 print(ranks)```

Output:
 0.1847  0.9904  0.1582
 0.6934  0.3652  0.8853
 0.5281  0.3054  0.7413
 0.5424  0.2891  0.4829
 0.2586  0.3230  0.8476
 0.6061  0.9546  0.3052
 0.5951  0.8214  0.7093
 0.1442  0.8960  0.9145
 0.6027  0.2672  0.0815
 0.3776  0.3473  0.5396
[torch.FloatTensor of size 10x3]

 2  3  1
 1  2  3
 1  2  3
 3  1  2
 1  2  3
 2  3  1
 1  2  3
 1  2  3
 1  2  3
 1  2  3
[torch.LongTensor of size 10x3]

Reply:
Oh yes, the order should be reversed.

Reply:
Thanks for pointing it out @satwikkottur, here is the test:

```
  1 require 'torch'
  2
  3 scores = torch.FloatTensor(10, 3):uniform();
  4
  5 sorted, ranks = scores:sort(2, true)
  6 print(scores)
  7 print(ranks)
```
```
 0.2974  0.6810  0.9164
 0.9497  0.2618  0.0178
 0.1139  0.3085  0.0496
 0.3462  0.6437  0.9521
 0.8896  0.7418  0.7638
 0.8688  0.5988  0.3765
 0.5683  0.8200  0.8018
 0.2770  0.3476  0.8543
 0.7989  0.4341  0.3507
 0.1155  0.4421  0.8378
[torch.FloatTensor of size 10x3]

 3  2  1
 1  2  3
 2  1  3
 3  2  1
 1  3  2
 1  2  3
 2  3  1
 3  2  1
 1  2  3
 3  2  1
[torch.LongTensor of size 10x3]
```